### PR TITLE
Refactoring PDB checks and adding additional usecase

### PR DESCRIPTION
**Pull request by ravitri**
### What type of PR is this?
_(bug/refactor)_

### What this PR does / why we need it?
This small PR addresses additional scenario of handling `MaxUnavailable: 0%`. The `if` condition became quite lengthy and tedious to read so the same was split into separate functions to scope the validations for `MaxUnavailable` and `MinAvailable`.

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ [OSD-25433](https://issues.redhat.com/browse/OSD-25433)

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster
- [x] Included documentation changes with PR

### CODE REVIEW

File: pkg/upgraders/healthcheck_pdb.go [hunk 1]
```diff
@@ -105,3 +106,38 @@ func checkDvoMetrics(c client.Client, dvo dvo.DvoClientBuilder, logger logr.Logg
...
+	if p.Spec.MaxUnavailable != nil {
+		switch p.Spec.MaxUnavailable.Type {
+		case intstr.Int:
+			if p.Spec.MaxUnavailable.IntVal == 0 {
+				l.Info(fmt.Sprintf("MaxUnavailable 0 found in PodDisruptionBudget: %s/%s\n", p.Namespace, p.Name))
+				return false, fmt.Errorf("found a PodDisruptionBudget with MaxUnavailable set to 0")
+			}
+		case intstr.String:
+			if p.Spec.MaxUnavailable.StrVal == "0%" {
+				l.Info(fmt.Sprintf("MaxUnavailable 0%% found in PodDisruptionBudget: %s/%s\n", p.Namespace, p.Name))
```
Comment (tkong-redhat) [inline pkg/upgraders/healthcheck_pdb.go:122]: Should be single % in the error log?

Comment (ravitri) [reply pkg/upgraders/healthcheck_pdb.go:122]: Hey Tony! Well, I wanted to show in logs that it's a percentage and not an integer so wanted to print % as a literal character. We need to have double %% to present it as a single % in logs. 

Refer https://pkg.go.dev/fmt:
> %%	a literal percent sign; consumes no value

File: pkg/upgraders/healthcheck_pdb.go [hunk 2]
```diff
@@ -105,3 +106,38 @@ func checkDvoMetrics(c client.Client, dvo dvo.DvoClientBuilder, logger logr.Logg
...
+}
+
+// validateMinAvailable function will return false for failures if
+// MinAvailable for a PDB is 100% if defined
+func validateMinAvailable(p policyv1.PodDisruptionBudget, l logr.Logger) (bool, error) {
+	if p.Spec.MinAvailable != nil {
+		switch p.Spec.MinAvailable.Type {
+		case intstr.String:
+			if p.Spec.MinAvailable.StrVal == "100%" {
+				l.Info(fmt.Sprintf("MinAvailable 100%% found in PodDisruptionBudget: %s/%s\n", p.Namespace, p.Name))
```
[the remark this answers is not in the document]
Comment (ravitri) [reply pkg/upgraders/healthcheck_pdb.go:137]: Same reply as previous comment

File: pkg/upgraders/healthcheck_pdb.go (hunk 1 above)
Comment (ravitri) [reply pkg/upgraders/healthcheck_pdb.go:122]: Example log:
```
4-09-19T05:52:56.564725387Z level=info logger=controller_upgradeconfig msg="MinAvailable 100% found in PodDisruptionBudget: ns1/ns1-pdb\n" Request.Namespace=openshift-managed-upgrade-operator Request.Name=managed-upgrade-config
```

File: pkg/upgraders/healthcheck_pdb.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkong-redhat) [reply pkg/upgraders/healthcheck_pdb.go:137]: Thank you. Sorry, I don't know about this. Knowledge extend.

**Pull request merged**